### DISCUSSION TITLE
mutate does not trigger onSuccess callback

### DISCUSSION THREAD

**Discussion by pke** [vercel/swr#759] [Q&A]
Is it intentionally that `mutate` will not trigger the `onSuccess` and `onError` callbacks for a hook?

```
const { data, mutate } = useSWR("/session", fetchSession, {
  onSuccess: newSession => newSession && AsyncStorage.setItem("session", newSession),
  onError: AsyncStorage.removeItem("session"),
})

const login = React.useCallback((name,password) => {
  return post(...).then(result => mutate(result, false))
})
```

**Comment by promer94** [comment]
mutate will not trigger the onSuccess and onError callbacks if you disable the revalidation process.

i think @sergiodxa 's [use-mutation](https://github.com/sergiodxa/use-mutation) is more suitable for your user case here
